app/main.py

```diff
@@ -1 +1,55 @@
-# write your code here
+class SoftwareEngineer:
+
+    def __init__(self, name):
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str):
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.skills += ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self):
+        print(f'{self.name} is creating a webpage...')
+        return '<h1>Hello world</h1>'
+
+
+class BackendDeveloper(SoftwareEngineer):
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.skills += ["Python", "SQL", "Django"]
+
+    def create_powerful_api(self):
+        print(f'{self.name} is creating an API...')
+        return 'http://127.0.0.1:8000'
+
+
+class AndroidDeveloper(SoftwareEngineer):
+
+    def __init__(self, name):
+        super().__init__(name)
+        self.skills += ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self):
+        print(f'{self.name} is creating a mobile app...')
+        return 'Ads every three swipes'
+
+
+class FullStackDeveloper(BackendDeveloper, FrontendDeveloper):
+
+    def __init__(self, name):
+        super().__init__(name)
+        FullStackDeveloper.skills = self.skills
+
+    def create_web_application(self):
+        print(f'{self.name} started creating a web application...')
+        back_developer = BackendDeveloper(self.name)
```

Comment:
You shouldn't create a new instance of the parent class from the child. You can call its methods directly because of inheritance
```
self.create_powerful_api()
```